src/GraphODataTemplateWriter/CodeHelpers/CSharp/TypeHelperCSharp.cs

```diff
@@ -431,5 +431,89 @@ public static string GetCoreLibraryType(this string @namespace, string type)
+        /// </summary>
+        /// <param name="methods">A list of OdcmMethods.</param>
+        /// <param name="namespace">The namespace for the methods.</param>
+        /// <returns>An ordered (to minimize diffs caused by reorder in the CSDL) list of 
+        /// MethodInfo objects used to construct the request builders.</returns>
+        public static List<MethodInfo> GetAllMethodsInfo(this List<OdcmMethod> methods, string @namespace)
+        {
+            return methods.Select(m =>
+            {
+                var parameters = m.Parameters
+                    .Select(p =>
+                    {
+                        var type = p.Type.GetTypeString(@namespace);
+                        var name = p.Name.ToLowerFirstChar();
+                        var parameterName = p.Name.GetSanitizedParameterName();
+                        var propertyName = p.Name.ToCheckedCase();
+
+                        // Adds support for classes ending in "Request" that have been dismabiguated.
+                        if (type.EndsWith("Request"))
+                        {
+                            type = String.Concat(type, "Object");
+                        }
+
+                        // Adjust the type string
+                        if (p.IsCollection)
+                        {
+                            type = $"IEnumerable<{type}>";
+                        }
+                        else if (!p.Type.IsTypeNullable() && p.IsNullable)
+                        {
+                            type += "?";
+                        }
+
+                        return new ParameterInfo(type, name, parameterName, propertyName, p.IsNullable);
+                    })
+                    .OrderBy(p => p.IsNullable ? 1 : 0); // I suspect this could cause a problem if new overloads are added.
+
+                var paramStrings = parameters.Select(p => string.Format(",\n            {0} {1}", p.Type, p.ParameterName));
+                var paramComments = parameters.Select(p => string.Format("\n        /// <param name=\"{0}\">A {0} parameter for the OData method call.</param>", p.ParameterName));
+                var paramArgsForConstructor = parameters.Select(p => string.Format(",\n                {0}", p.ParameterName));
+
+                var entityName = m.Class.Name.ToCheckedCase();
+                var methodName = m.Name.ToCheckedCase();
+                var requestType = entityName + methodName + "Request";
+                var requestBuilderType = requestType + "Builder";
+
+                return new MethodInfo(
+                    parameters,
+                    string.Join("", paramStrings),
+                    string.Join("", paramArgsForConstructor),
+                    string.Join("", paramComments),
+                    requestBuilderType,
+                    methodName,
+                    m.FullName,
+                    paramStrings.Count() == 0 ? "" : string.Join("", paramStrings).Substring(1)
```

Comment:
I'm guessing the substring is because you have a space in front you don't want, comment?

Reply:
We don't want the leading ',' from ",\n            {p.Type} {p.ParameterName}"

Reply:
have you considered TrimStart(',') to make it more explicit then?